configs/envoy-demo.yaml

```diff
@@ -55,3 +55,8 @@ static_resources:
       typed_config:
         "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
         sni: www.envoyproxy.io
+        common_tls_context:
+          validation_context:
+            trusted_ca:
+              filename: certs/upstreamcacert.pem
```

Comment:
Does this actually work? Doesn't this need to use the system root chain? Same elsewhere.

Reply:
also need a `match_subject_alt_names` to be a good demo

Reply:
Verifying SAN is getting at my point in the linked issue which is that "verifying TLS" is a bit tricky. Obviously verifying the cert against a chain is required, but beyond that there is SAN verification, pinning, etc. and it seems like we might want to enforce different things in different contexts, or for example warn if one of SAN verification, pinning, etc. is not configured.

Reply:
I think I'd rather do any changes we're going to do here in one fell swoop, so let's sort out what we think the correct behavior is.
I think making sure we have a valid cert chain is pretty key.  I don't love having to manually  configure match_subject_alt_names.  I would think we'd want to make sure that the host headers sent on a given connection match the SAN for that connection, but I don't see that as a configuration option.

Reply:
See `auto_san_validation`. The problem here is that this obviously only works for HTTP, and this configuration is generic for all protocols.

This is why I said this is really tricky. :)

I think what you want is something like:
1) Valid cert chain is on by default
2) At least one of a) cert pinning, b) matching a specific SAN, c) auto SAN, etc. is configured.

Reply:
I did that list of 3 from memory. Please take a look and see if anything else should be in there. cc @ggreenway who knows more about this type of stuff also.

Reply:
coming back to this, I'm a bit concerned that we'll still have a gap here, where if we allow clusters with http configuration to specify auto_san, but then don't have pinned certificates or a default san that we won't actually verify anything for TCP traffic.  Should we require either pinning or a default san just to handle this case?